atmcfg/atmcfg.go

```diff
@@ -72,11 +78,45 @@ func setDisabledByShardName(out *opsmngr.AutomationConfig, name string, disabled
 	}
 }
 
+func setDisableProcessByProcessNameAndPort(out *opsmngr.AutomationConfig, processName string, processPortNum int, disabled bool) {
+	newDeploymentAuthMechanisms(out)
+	for k, p := range out.Processes {
+		if p.Hostname == processName && p.Args26.NET.Port == processPortNum {
+			out.Processes[k].Disabled = disabled
+		}
+	}
+}
+
 // Shutdown disables all processes of the given cluster name.
 func Shutdown(out *opsmngr.AutomationConfig, name string) {
 	setDisabledByClusterName(out, name, true)
 }
 
+// ShutdownProcess disables processes. Processes are provided in the format {"hostname:port","hostname2:port2"}.
+func ShutdownProcess(out *opsmngr.AutomationConfig, processes []string) error {
```

Comment:
[nit] consider that what's on the spec is process per cluster and this is a list of process not looking at the cluster name, it was my expectation to change `Shutdown` to be `Shutdown(out *opsmngr.AutomationConfig, name string, ...processes)` which keeps it backward compatible where no process means all but feel free to add a new method just this one doesn't look at the cluster name